src/gui/filedetails/sharemodel.h

```diff
@@ -196,8 +197,8 @@ private slots:
     // We cannot pass objects instantiated in QML using smart pointers through the property interface
     // so we have to pass the pointer here. If we kill the dialog using a smart pointer then
     // these objects will be deallocated for the entire application. We do not want that!!
-    AccountState *_accountState;
-    Folder *_folder;
+    QPointer<AccountState> _accountState;
+    QPointer<Folder> _folder;
```

Comment:
is this change really needed ?
can those objects be deleted during the application lifetime ?

Reply:
I could ask the opposite -- what's the harm in adding this change? 

Looking at the possible crash paths for `ShareModel::sharingEnabled` I see that the only situation where it makes sense is that we have a dangling `_accountState` pointer, which a `QPointer` would prevent a bad access to

`_accountState->account()` is a smart pointer and `capabilities` is not a pointer